src/main/java/games/strategy/engine/config/PropertyFileReader.java

```diff
@@ -0,0 +1,56 @@
+package games.strategy.engine.config;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Properties;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Throwables;
+
+/**
+ * Reads property values from the game engine configuration file.
+ * 
+ * @see PropertyReader for a complete listing of property keys
+ */
+public class PropertyFileReader implements PropertyReader {
+
+  private final File propertyFile;
+
+  PropertyFileReader(String propertyFile) {
+    this(new File(propertyFile));
+  }
+
+  @VisibleForTesting
+  PropertyFileReader(final File propertyFile) {
+    this.propertyFile = propertyFile;
+  }
+
+
+  protected String readProperty(final String propertyKey) {
+    try (FileInputStream inputStream = new FileInputStream(propertyFile)) {
+      final Properties props = new Properties();
+      props.load(inputStream);
+
+      if (!props.containsKey(propertyKey)) {
+        throw new PropertyNotFoundException(propertyKey, propertyFile.getAbsolutePath());
+      } else {
+        return props.getProperty(propertyKey).trim();
+      }
+    } catch (final FileNotFoundException e) {
+      throw Throwables.propagate(e);
+    } catch (final IOException e) {
+      throw new IllegalStateException("Failed to read propertyFile: " + propertyFile.getAbsolutePath(), e);
+    }
+  }
+
+  static  class PropertyNotFoundException extends IllegalStateException {
```

Comment:
Should this class be `public` since it seems to be part of the `readProperty()` API, and `readProperty()` can be called by classes outside this package?

I'm not sure about the constructor accessibility.  It all comes down to whether or not subclasses in other packages (of which there are currently none) that override `readProperty()` need to be able to throw instances of this class, as well.